src/Config.php

```diff
@@ -43,11 +44,14 @@ public function __construct($target, $conf = null, CacheItemPoolInterface $cache
             $this->gcp_call_invoker = new \Grpc\DefaultCallInvoker();
             return;
         }
+
         $gcp_channel = null;
         $url_host = parse_url($target, PHP_URL_HOST);
         $this->hostname = $url_host ? $url_host : $target;
         $channel_pool_key = $this->hostname . '.gcp.channel.' . getmypid();
 
+        $this->cross_script_shmem_enabled = $this->crossScriptShmemEnabled();
```

Comment:
What is the reason for moving this method invocation out of the callable?

Reply:
Hey John can you share more details? Did you mean I should assign $this->cross_script_shmem_enabled inside crossScriptShmemEnabled() ?

Reply:
The previous commit didn't call `crossScriptShmemEnabled()` until line 72 (within the `register_shutdown_function()` callable). I was just curious why you moved the call out of the callable and assigned it to an object property, as it's not used anywhere else besides the shutdown function.

Reply:
Since this is a fixed value and will not get changed during run time, I just thought it would be better to assign it to a private variable, in case in future we need to check the boolean in some other place, we don't need to call the function again. I can sure create another commit if you think the previous way is better :)

Reply:
I can see that it could very likely be needed again in the future, so the property is ok with me. :)